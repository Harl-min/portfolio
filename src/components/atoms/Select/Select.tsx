import classNames from 'classnames';

export interface SelectProps {
	label: string;
	id: string;
	isRequired?: boolean;
	error?: string;
	placeholder?: string;
	[rest: string]: unknown;
	options: Array<{
		value: string;
		label: string;
	}>;
}

export const Select = ({
	label,
	id,
	isRequired,
	error,
	options,
	placeholder,
	...rest
}: SelectProps) => {
	return (
		<div className="flex flex-col gap-1">
			<label htmlFor={id} className="font-bold">
				{label} {isRequired && <span className="text-red-700">*</span>}
			</label>
			<select
				id="subject"
				className={classNames(
					'my-2 rounded-md px-4 py-2 ring-1 dark:bg-transparent',
					{
						'ring-red-600': !!error,
						'ring-grey-400 dark:ring-slate-500': !error,
					}
				)}
				{...rest}
			>
				{placeholder && (
					<option value="" disabled>
						{placeholder}
					</option>
				)}
				{options.map(({ value, label }) => (
					<option key={value} value={value}>
						{label}
					</option>
				))}
			</select>
			{error && (
				<div className="font-sm font-normal -mt-2 text-red-600">{error}</div>
			)}
		</div>
	);
};
