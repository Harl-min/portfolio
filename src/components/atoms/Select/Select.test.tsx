import { Select } from '../Select';

import { render } from '@testing-library/react';

const mockField = {
	value: '',
	checked: false,
	onChange: jest.fn(),
	onBlur: jest.fn(),
	multiple: undefined,
	name: 'password',
};
const mockMeta = {
	touched: false,
	error: '',
	initialError: '',
	initialTouched: false,
	initialValue: '',
	value: '',
};

describe('Select', () => {
	it('renders correctly', () => {
		const { container } = render(
			<Select label="Some select" id="some-select" options={['1', '2', '3']} />
		);
		expect(container).toMatchSnapshot();
	});
});
