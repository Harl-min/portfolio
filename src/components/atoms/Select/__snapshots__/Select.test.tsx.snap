// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select renders correctly 1`] = `
<div>
  <div
    class="flex flex-col gap-1"
  >
    <label
      class="font-bold"
      for="some-select"
    >
      Some select
       
    </label>
    <select
      class="my-2 rounded-md px-4 py-2 ring-1 dark:bg-transparent ring-grey-400 dark:ring-slate-500"
      id="subject"
    >
      <option />
      <option />
      <option />
    </select>
  </div>
</div>
`;
